### ISSUE TITLE
Cannot figure out how to build OpenEXR under mingw64 with v2.3.0

### ISSUE THREAD

**Issue by jimdinunzio** [AcademySoftwareFoundation/openexr#340]
I am using MSYS2 with mingw64 and Qt.  I was able to build OpenEXR with the previous version 2.2.1 with some minor modifications using the CMake build route, but with v2.3.0 I managed to build static libraries for IlmBase after half a day, but with the rest of the day could not get OpenEXR to even create its CMake files without errors.   I guess I'll stick with the older version for now to do my research.
Upgrading would be nice eventually. Any help would be appreciated.
Jim

Running "C:\msys64\mingw64\bin\cmake.exe C:/Users/jim.dinunzio/Documents/git/ampas/openexr/OpenEXR_Viewers "-GCodeBlocks - MinGW Makefiles" "-DCMAKE_BUILD_TYPE:STRING=Release" "-DCMAKE_CXX_COMPILER:STRING=C:/Qt/qt-5.4.2-x64/mingw64/bin/g++.exe" "-DCMAKE_INSTALL_PREFIX:PATH=c:/msys64/mingw64" "-DQT_QMAKE_EXECUTABLE:STRING=C:/Qt/qt-5.4.2-x64/qt-5.4.2-x64-mingw510r0-seh-opengl-rev0/bin/qmake.exe"" in C:\Users\jim.dinunzio\AppData\Local\Temp\qtc-cmake-OvGsOW.
ILMBASE_PACKAGE_PREFIX = C:/msys64/mingw64
CMake Error at IlmImf/CMakeLists.txt:19 (ADD_EXECUTABLE):
  Target "b44ExpLogTable" links to target "IlmBase::Half_static" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at IlmImf/CMakeLists.txt:19 (ADD_EXECUTABLE):
  Target "b44ExpLogTable" links to target "IlmBase::IlmThread_static" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at IlmImf/CMakeLists.txt:19 (ADD_EXECUTABLE):
  Target "b44ExpLogTable" links to target "IlmBase::Iex_static" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at IlmImf/CMakeLists.txt:51 (ADD_EXECUTABLE):
  Target "dwaLookups" links to target "IlmBase::Half_static" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at IlmImf/CMakeLists.txt:51 (ADD_EXECUTABLE):
  Target "dwaLookups" links to target "IlmBase::IlmThread_static" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

CMake Error at IlmImf/CMakeLists.txt:51 (ADD_EXECUTABLE):
  Target "dwaLookups" links to target "IlmBase::Iex_static" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at IlmImfExamples/CMakeLists.txt:3 (ADD_EXECUTABLE):
  Target "IlmImfExamples" links to target "OpenEXR::IlmImf" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at IlmImfExamples/CMakeLists.txt:3 (ADD_EXECUTABLE):
  Target "IlmImfExamples" links to target "IlmBase::Half" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at IlmImfExamples/CMakeLists.txt:3 (ADD_EXECUTABLE):
  Target "IlmImfExamples" links to target "IlmBase::Iex" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at IlmImfExamples/CMakeLists.txt:3 (ADD_EXECUTABLE):
  Target "IlmImfExamples" links to target "IlmBase::Imath" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at IlmImfExamples/CMakeLists.txt:3 (ADD_EXECUTABLE):
  Target "IlmImfExamples" links to target "IlmBase::IlmThread" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

-- Configuring done
Running "C:\msys64\mingw64\bin\cmake.exe C:/Users/jim.dinunzio/Documents/git/ampas/CTL "-GCodeBlocks - MinGW Makefiles" "-DCMAKE_BUILD_TYPE:STRING=Release" "-DCMAKE_CXX_COMPILER:STRING=C:/Qt/qt-5.4.2-x64/mingw64/bin/g++.exe" "-DCMAKE_INSTALL_PREFIX:PATH=c:/msys64/mingw64" "-DQT_QMAKE_EXECUTABLE:STRING=C:/Qt/qt-5.4.2-x64/qt-5.4.2-x64-mingw510r0-seh-opengl-rev0/bin/qmake.exe"" in C:\Users\jim.dinunzio\Documents\git\ampas\build-CTL-Qt_5_4_2_64_bits-Release.
-- Generating done
-- Build files have been written to: C:/Users/jim.dinunzio/Documents/git/ampas/build-IlmBase-Qt_5_4_2_64_bits-Release
-- Generating done
-- Build files have been written to: C:/Users/jim.dinunzio/Documents/git/ampas/build-OpenEXR-Qt_5_4_2_64_bits-Release
-- The C compiler identification is GNU 5.1.0
-- found IlmBase, version 2.2.0
-- found TIFF, version 4.0.8
-- Found OpenEXR, version 2.2.0
-- Found AcesContainer, version 1.0.2
*** cmake process exited with exit code 1.
-- The CXX compiler identification is GNU 5.1.0
-- Check for working C compiler: C:/Qt/qt-5.4.2-x64/mingw64/bin/gcc.exe
-- Check for working C compiler: C:/Qt/qt-5.4.2-x64/mingw64/bin/gcc.exe -- works
-- Detecting C compiler ABI info
CMake Warning (dev) at CMakeLists.txt:87 (install):
  Policy CMP0062 is not set: Disallow install() of export() result.  Run
  "cmake --help-policy CMP0062" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The file

    C:/Users/jim.dinunzio/Documents/git/ampas/build-CTL-Qt_5_4_2_64_bits-Release/CTLLibraryDepends.cmake

  was generated by the export() command.  It should not be installed with the
  install() command.  Use the install(EXPORT) mechanism instead.  See the
  cmake-packages(7) manual for more.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/jim.dinunzio/Documents/git/ampas/build-CTL-Qt_5_4_2_64_bits-Release
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Qt/qt-5.4.2-x64/mingw64/bin/g++.exe
-- Check for working CXX compiler: C:/Qt/qt-5.4.2-x64/mingw64/bin/g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
OPENEXR_PACKAGE_PREFIX = 
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    QT_QMAKE_EXECUTABLE

**Comment by meshula** [comment]
I think you are building the viewers, from within the viewers directory?

First off, there's a bug in the cmake scripts that prevents the viewers from linking; it's corrected in this pull request: https://github.com/openexr/openexr/pull/218

Second, building cmake components individually from their directories was the supported path in 2.2. At the moment, building OpenEXR from the root cmake file is recommended. We are working through restoring the ability of building from the subdirectories, but it's not quite ready. Sorry for the hassle.

**Comment by jimdinunzio** [comment]
Hi Meshula,
Thanks.  I was not building from the root folder.  That helped a lot.  Although I still spent a day figuring out issues building it and building ACES CTL which was my ultimate goal.  I found the fix for the viewers and applied it, thanks.  Here are my check-in notes on my local branch msys64-mingw64.  This is in addition to the porting changes I made last December.

- Apply fix to viewer build from current pull request in develop branch
- Turn off shared libraries because they are not building and other unneeded options. (gnu compiler doesn't work well with dllimport with inline functions?)  (got several of these errors from IexBaseExe.h macro Error: function definition is marked dllimport )
- remove _s suffix from static library filenames because CTL can't find them otherwise.  
- add definitions (-D_USE_MATH_DEFINES) due to compile error
